app/models/signup.rb

```diff
@@ -0,0 +1,42 @@
+class Signup
+
+  extend ActiveModel::Naming
+  include ActiveModel::Conversion
+  include ActiveModel::Validations
+
+  attr_reader :user, :company, :first_name, :last_name, :company_name, :email, :password
+
+  # Forms are never themselves persisted
+  def initialize(params)
+    @first_name = params[:user][:first_name]
+    @last_name = params[:user][:last_name]
+    @company_name = params[:user][:company][:company_name]
+    @email = params[:user][:email]
+    @password = params[:user][:password]
+  end
+
+  def persisted?
+    false
+  end
+
+  def save
+    if valid?
```

Comment:
How this validation is checked? What fields are taken into account and in what format?